tests/test_vega.py

```diff
@@ -263,7 +256,776 @@ def test_fill_anchor_in_string(tmp_dir):
+    pivot_field,
+    group_by,
+):  # pylint: disable=too-many-arguments
+    props = {
+        "template": "linear",
+        "x": "step",
+        "y": y,
+        "revs_with_datapoints": ["B"],
+        "anchors_y_definitions": anchors_y_definitions,
+    }
+
+    expected_datapoints = _get_expected_datapoints(datapoints, expected_dp_keys)
+
+    renderer = VegaRenderer(datapoints, "foo", **props)
+    plot_content = renderer.get_filled_template()
+
+    assert plot_content["data"]["values"] == expected_datapoints
+    assert plot_content["encoding"]["color"] == {
+        "field": "rev",
+        "scale": {"domain": ["B"], "range": ["#945dd6"]},
+    }
+    assert plot_content["encoding"]["strokeDash"] == stroke_dash_encoding
+    assert plot_content["layer"][3]["transform"][0]["calculate"] == pivot_field
+    assert plot_content["layer"][0]["transform"][0]["groupby"] == group_by
+
+
+@pytest.mark.parametrize(
+    ",".join(
+        [
+            "datapoints",
+            "y",
+            "anchors_y_definitions",
+            "expected_dp_keys",
+            "row_encoding",
+            "group_by_y",
+            "group_by_x",
+        ]
+    ),
+    (
+        (
+            [
+                {
+                    "rev": "B",
+                    "predicted": "0.05",
+                    "actual": "0.5",
+                    "filename": "test",
+                    "field": "predicted",
+                },
+                {
+                    "rev": "B",
+                    "predicted": "0.9",
+                    "actual": "0.9",
+                    "filename": "test",
+                    "field": "predicted",
+                },
+            ],
+            "predicted",
+            [{"filename": "test", "field": "predicted"}],
+            ["rev", "predicted", "actual"],
+            {},
+            ["rev", "predicted"],
+            ["rev", "actual"],
+        ),
+        (
+            [
+                {
+                    "rev": "B",
+                    "predicted": "0.05",
+                    "actual": "0.5",
+                    "filename": "test",
+                    "field": "predicted",
+                },
+                {
+                    "rev": "B",
+                    "predicted": "0.9",
+                    "actual": "0.9",
+                    "filename": "train",
+                    "field": "predicted",
+                },
+            ],
+            "predicted",
+            [
+                {"filename": "test", "field": "predicted"},
+                {"filename": "train", "field": "predicted"},
+            ],
+            ["rev", "predicted", "actual"],
+            {"field": "filename", "sort": []},
+            ["rev", "filename", "predicted"],
+            ["rev", "filename", "actual"],
+        ),
+        (
+            [
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.05",
+                    "predicted_test": "0.05",
```

Comment:
Seems like it might be good to have a couple more cases here:

1. Same filename but different field names
2. Different filenames but same field name

Reply:
> Why predicted_test instead of predicted? Is the filename not enough to group these?

This particular test is for the `filename::field` variation of the confusion matrices template. The y definitions are:

```
{"filename": "test", "field": "predicted_test"},
{"filename": "train", "field": "predicted_train"},
```


There are 4 different available splits for each plot:

1. single source
2. same file, different fields (field)
3. same field, different files (filename)
4. different file, different fields (filename::field)

The logic for 2/3 is very similar. So for each group of tests (I thought) I've added an example of 1 & (2 or 3) & 4. I will expand to have a test for each of the 4 cases. 

The reason that I didn't just "test everything" is that I felt that the volume of code was already beyond a size which I could reasonably ask to be reviewed.

Reply:
[F] Now each of the new tests should have a case for each of the 4 variants

1 = "single_source"
2 = "multi_field"
3 = "multi_filename"
4 = "multi_filename_field"

Reply:
Thanks @mattseddon! I think having all 4 makes it easier to read even though it's longer since you have the same structure for each test.